### DISCUSSION TITLE
Child Resource

### DISCUSSION THREAD

**Discussion by invaders-xx** [filamentphp/filament#3118] [Help]
Is there a way to create a resource which is a relation to another resource ?
Parent has many children, create a ChildRessource which is related to Parent's id ?
Many thanks for any tips&tricks

**Comment by invaders-xx** [comment]
@danharrin, do you have any suggestions ? Many thanks in advance.

David

**Reply by ryangjchandler** [reply]
Surely you just want a `Select` that handles the `belongsTo` relationship, no? https://filamentphp.com/docs/2.x/admin/resources/getting-started#belongsto

**Reply by invaders-xx** [reply]
No I don't. In fact I want a Resource that will be accessible from parent resource's list via a table's button, and display related data.
It's mostly like a RelationManager but in a "Resource Mode" view.
I hope I made myself clear enough, sorry for my english.

**Reply by danharrin** [reply]
Like https://filamentphp.com/tricks/relation-manager-create-record-without-modal and https://filamentphp.com/tricks/relation-manager-edit-record-without-modal ?

**Reply by invaders-xx** [reply]
Not really in fact. Imagine a model A which has many Model B. Instead of having a relation manager in A for B, I would like to have a resource for A and a resource for B. The resource for B is only accessible from resource A record's list with a button to access resource B record's list and I would like to know how to make resource B dependant to resource A id ?

**Reply by invaders-xx** [reply]
any idea @danharrin ?

**Reply by danharrin** [reply]
No idea, except using the forms and tables packages to build your own structure in custom pages

**Comment by moskoweb** [comment]
I'm also in need of a structure of this type, I think it could be done something like in Resource B being able to have a definition of "parent", which could be defined in getPages, as below:

```
public static function getPages(): array
    {
        return [
            'index' => Pages\ListModels::route('/{parent}/'),
            'create' => Pages\CreateModel::route('/{parent}/create'),
            'view' => Pages\ViewModel::route('/{parent}/{record}'),
            'edit' => Pages\EditModel::route('/{parent}/{record}/edit'),
        ];
    }
```

Also defining a "parentColumn", so that it can be used in "getEloquentQuery", to fetch information from model A.

```
protected static ?string $parentColumn = 'model_a_id';
```

So in "getEloquentQuery", you could do a ```where($parentColumn, $parent)```.

**Comment by kaitwalla** [comment]
Create a field on the resource constrained to a foreign key on itself.

```
Schema::table('resource', function (Blueprint $table) {
            $table->foreignId('parent_id')->nullable()->constrained('resource')->onDelete('cascade');
        });
```

Use a default Eloquent one-to-one relationship:

```
 public function parent()
    {
        return $this->belongsTo(self::class, 'parent_id');
    }
```

Use a select field with the `relationship()` method in your builder:
```
Select::make('parent_id')
                ->label('Parent')
                ->relationship('parent', 'title')
                ->searchable(),
```

Add the following to the model if you ever need to get all the children:

```
 public function children()
    {
        return self::where('parent_id', $this->id)->get();
    }
 ```

**Comment by eclipxe13** [comment]
I think that article [Filament v3 Nested Resources: Solution with Trait and Pages](https://laraveldaily.com/post/filament-v3-nested-resources-trait-pages) from the people from Laravel Daily goes right into this issue.
I was searching for the same problem (using infolist wasn't the right option at this time) and found the article.
